bin/fork

```diff
@@ -9,41 +9,93 @@
 # into.  Default is the name of the repo relative to the current working
 # directory.
 
-GH_USERNAME=`grep user ~/.config/hub | tail -1 | sed "s/- user: //"`
-GH_TOKEN=`grep oauth_token ~/.config/hub | tail -1 | sed "s/  oauth_token: //"`
+# initial variable defaults to github
+# or whatever GITHOST env is set to
 
-FORKED_ORG_REPO=$1
-FORKED_ORG=${FORKED_ORG_REPO%/*}
-FORKED_REPO=${FORKED_ORG_REPO#*/}
-FORKED_DIR=${2-${FORKED_ORG_REPO#*/}}
-FORKED_URL="https://github.com/$FORKED_ORG/$FORKED_REPO.git"
-NEW_FORK_URL="https://github.com/$GH_USERNAME/$FORKED_REPO"
-IS_FORKED=$(curl --silent -o /dev/null --fail --write-out "%{http_code}" "$NEW_FORK_URL")
+GITHOST=${GITHOST:-github.com}
+
+# function
 
 ohai() {
   echo -e "\033[34m==>\033[0m \033[1m$1\033[0m"
 }
 
-if [[ $IS_FORKED == "404" ]]; then
-  ohai "Forking $FORKED_URL ..."
-  curl -sH "Authorization: token $GH_TOKEN" -X POST \
-       https://api.github.com/repos/$FORKED_ORG/$FORKED_REPO/forks > /dev/null
+# parse options
+
+while [ True ]; do
+if [ "$1" = "--help" -o "$1" = "-h" ]; then
+    echo " "
+    echo "$0 [--host] REPO"
+    echo " "
+    exit
+elif [ "$1" = "--host" -o "$1" = "-h" ]; then
+    GITHOST="${2}"
+    shift 2
+elif [ "$1" = "--verbose" -o "$1" = "-v" ]; then
+    VERBOSE=1
+    shift 1
+else
+    break
+fi
+done
+
+# if anything is left, it is NAMESPACE/REPO to be forked
+# followed by the desired name of new FORKED_DIR
+
+GITHOST_CONFIG=${GITHOST_CONFIG:-$HOME/.config/githost}
+
+GITHOST_USER=`sed -nr "/^\[$GITHOST\]/ { :l /^user[ ]*=/ { s/.*=[ ]*//; p; q;}; n; b l;}" "${GITHOST_CONFIG}"`
+GITHOST_TOKEN=`sed -nr "/^\[$GITHOST\]/ { :l /^token[ ]*=/ { s/.*=[ ]*//; p; q;}; n; b l;}" "${GITHOST_CONFIG}"`
 
-  # give github 5 seconds before cloning the newly forked repo
-  sleep 5
+FORKED_ORG_REPO="${1}"
+FORKED_ORG=${FORKED_ORG_REPO%/*}
+FORKED_REPO=${FORKED_ORG_REPO#*/}
+
+FORKED_DIR=${2-${FORKED_ORG_REPO#*/}}
+FORKED_URL="$FORKED_ORG/$FORKED_REPO.git"
+NEW_FORK_URL="$GITHOST_USER/$FORKED_REPO"
+IS_FORKED=$(curl --silent -o /dev/null --fail --write-out "%{http_code}" "${GITHOST}"/"$NEW_FORK_URL")
+
+if [[ "$VERBOSE == 1" ]]; then
+echo "HOST: $GITHOST"
+echo "USER: $GITHOST_USER"
+echo "ORG: $FORKED_ORG"
+echo "REPO: $FORKED_REPO"
+echo "TOKEN: $GITHOST_TOKEN"
+echo "DEST_DIR: $FORKED_DIR"
+fi
+
+# main
+
+if [[ "$IS_FORKED" != "404" && "$GITHOST" == "github.com" ]]; then
```

Comment:
Added

Reply:
Thanks, fixed. I believe your script might be failing sometimes, too. I realised after further testing that http calls to github return 301 (redirecting to https). Haven't verified that with rigorous testing, but it would explain some odd things I'd noticed early on (or it could have all been user error)